### ISSUE TITLE
This question status

### ISSUE THREAD

**Issue by privetpoka** [owlkaboom/quarkus-threading-question#1]
Hello. I've noticed the similar behavior, but from another angle: When I run some verticle as a separate worker in dev mode it runs in worker thread and in production build (both uber jar and from Quarkus-app folder it runs as executor. 
Do you know if it can be fixed or is it somewhat of expected behavior? 
Thanks.

**Comment by owlkaboom** [comment]
I opened the following issue on the quarkus github. https://github.com/quarkusio/quarkus/issues/19911#issue-987991327

Looks like they have determined this behavior as expected. I cannot share the implementation code on our "work around" but as a simple example we throttled our producer which was pumping work into an event bus.

For example we would only push ~20 events into this event bus and every time an event was fully processed we would decrement a "shared" counter so the producer can continue to push more events until everything has been fully consumed.

**Comment by privetpoka** [comment]
Thank you for the clarification. 

I understand, also I noticed that if something annotated as @Scheduled in dev env it is executed in executor-worker, though, in prod build it runs in the same vertx event loop threads.
 
It actually impacts performance as we notices that during scheduled operations (which run in event-loop in production) there are spikes in 5XX error due to schedulers occupy the same thread-pool. 

So FYI if you use Scheduled annotation in quarkus it might affect performance. 
As a workaround we moved schedulers to works as verticles.  Not ideal, but it works. Still wondering why vertx-worker are not available..
